mlrun/runtimes/base.py

```diff
@@ -1018,11 +1033,7 @@ def _is_runtime_resource_run_in_transient_state(
         now = datetime.now(timezone.utc)
         last_update_str = run.get('status', {}).get('last_update', now)
         last_update = datetime.fromisoformat(last_update_str)
-        if (
-            last_update
-            + timedelta(seconds=float(config.runtime_resources_deletion_grace_period))
-            > now
-        ):
+        if last_update + timedelta(seconds=float(grace_period)) > now:
```

Comment:
Why here and not by whatever uses this? the grace period doesn't effect the state of the object, it's just another input to take into account.
Maybe a restriction w.r.t `last_update` being available only here?